### PR TITLE
test: increase transfer test coverage

### PR DESCRIPTION
**Pull request by steffenix**
@fubuloubu does that look like what you need help on regarding increasing test coverage on #158 ?

### CODE REVIEW

File: contracts/test/TokenWithBlackList.sol [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    function isBlackListed(address user) public view returns (bool) {
+        return _blacklist[user];
+    }
+
+    function _beforeTokenTransfer(
+        address from,
+        address to,
+        uint256 amount
+    ) internal virtual override(ERC20) {
+        require(!isBlackListed(to), "Token transfer refused. Receiver is on blacklist");
```
Comment (fubuloubu) [inline contracts/test/TokenWithBlackList.sol:26]: Can we add this funxtionality to our test token instead? (Rename to `public mapping (address => bool) _blocked` and `_setBlocked(address, bool)`)

Comment (fubuloubu) [reply contracts/test/TokenWithBlackList.sol:26]: No reason to create an entire new token for one testing feature

File: tests/functional/vault/test_withdrawal.py [hunk 2]
```diff
@@ -259,3 +259,27 @@ def test_withdrawal_with_reentrancy(
...
+    vault.setDepositLimit(2 ** 256 - 1, {"from": gov})
+    starting_balance = tokenWithBlackList.balanceOf(gov)
+
+    tokenWithBlackList.approve(vault, 2 ** 256 - 1, {"from": gov})
+    vault.deposit({"from": gov})
+
+    tokenWithBlackList.blacklistUpdate(gov, True, {"from": gov})
+
+    with brownie.reverts():
+        vault.withdraw(vault.balanceOf(gov), {"from": gov})
```
Comment (fubuloubu) [inline tests/functional/vault/test_withdrawal.py:285]: With the change above, this is now what the test can be refactored to

File: contracts/test/TokenWithBlackList.sol (hunk 1 above)
Comment (steffenix) [reply contracts/test/TokenWithBlackList.sol:26]: Sure I will do it

File: tests/functional/vault/test_withdrawal.py (hunk 2 above)
Comment (steffenix) [reply tests/functional/vault/test_withdrawal.py]: I do not see suggested changes.

File: tests/functional/vault/test_withdrawal.py [hunk 3]
```diff
@@ -259,3 +259,25 @@ def test_withdrawal_with_reentrancy(
     # given previous setup the withdraw should revert from reentrancy guard
     with brownie.reverts():
         vault.withdraw(vault.balanceOf(gov), {"from": gov})
+
+
+def test_token_transfer_fail(token, gov, Vault):
```
Comment (fubuloubu) [inline tests/functional/vault/test_withdrawal.py:264]: Not necessary if merged into previous test

File: tests/functional/vault/test_misc.py [hunk 4]
```diff
@@ -124,3 +124,23 @@ def test_reject_ether(gov, vault):
...
+        token,
+        gov,
+        gov,
+        token.symbol() + " yVault",
+        "yv" + token.symbol(),
+        gov,
+        {"from": gov},
+    )
+    vault.setDepositLimit(2 ** 256 - 1, {"from": gov})
+    token.approve(vault, 2 ** 256 - 1, {"from": gov})
```
[the remark this answers is not in the document]
Comment (fubuloubu) [reply tests/functional/vault/test_misc.py:142]: Not sure I understand why the comment `Need a fresh vault to do this math right` is there

File: tests/functional/vault/test_withdrawal.py (hunk 3 above)
Comment (steffenix) [reply tests/functional/vault/test_withdrawal.py:264]: Understood!

File: tests/functional/vault/test_withdrawal.py [hunk 5]
```diff
@@ -259,3 +259,25 @@ def test_withdrawal_with_reentrancy(
...
+        {"from": gov},
+    )
+    vault.setDepositLimit(2 ** 256 - 1, {"from": gov})
+    token.approve(vault, 2 ** 256 - 1, {"from": gov})
+    vault.deposit({"from": gov})
+
+    token._setBlocked(gov, True, {"from": gov})
+
+    with brownie.reverts():
+        vault.withdraw(vault.balanceOf(gov), {"from": gov})
```
Comment (fubuloubu) [inline tests/functional/vault/test_withdrawal.py:283]: This entire test should go away now that you merged the functionality into the other one

**Review by fubuloubu — APPROVED** [review]

**Comment by fubuloubu** [comment]
Awesome, will merge once tests pass

**Comment by steffenix** [comment]
Do you need multiple approval to merge?

**Pull request merged**